### ISSUE TITLE
overlay broken (appears white) as of last update

### ISSUE THREAD

**Issue by abc-mikey** [ValveSoftware/steam-for-linux#7169]
#### Your system information

```
Steam client application
Built: Jun 4 2020, at 05:51:01
Steam API: v020
Steam package versions: 1591251555
```

```
NVIDIA Driver Version: 440.82
```

Distribution: Fedora31
Opted into Steam client beta?: No
Have you checked for system updates: Yes

#### Please describe your issue in as much detail as possible:

The steam overlay has been broken in big picture mode (using the steamcompositor) since the last update I received. Info boxes and accessing it through [shift]+[tab] appear as a white box / white screen respectively. 

#### Steps for reproducing this issue:

1. Start big picture mode session
2. Start game
3. [shift]+[tab]

[log_steam3.txt](https://github.com/ValveSoftware/steam-for-linux/files/4740121/log_steam3.txt)
[steam_client.txt](https://github.com/ValveSoftware/steam-for-linux/files/4740133/steam_client.txt)
[steam_packages.txt](https://github.com/ValveSoftware/steam-for-linux/files/4740134/steam_packages.txt)
[usr_bin_steamon-session.txt](https://github.com/ValveSoftware/steam-for-linux/files/4740135/usr_bin_steamon-session.txt)
[vulkaninfo2.txt](https://github.com/ValveSoftware/steam-for-linux/files/4740165/vulkaninfo2.txt)

**Comment by abc-mikey** [comment]
[steam_client.txt](https://github.com/ValveSoftware/steam-for-linux/files/4740112/steam_client.txt)

**Comment by alkazar** [comment]
I and a few other folks are also seeing this on GamerOS. It started happening with the June 1 update and only on Nvidia.

**Comment by alkazar** [comment]
In case it helps anyone, I was able to partially restore the overlay functionality with the following patch to the SteamOS compositor:
```
diff --git a/src/steamcompmgr.c b/src/steamcompmgr.c
index aa3a520..0faf025 100644
--- a/src/steamcompmgr.c
+++ b/src/steamcompmgr.c
@@ -508,7 +509,7 @@ ensure_win_resources (Display *dpy, win *w)
        if (!w->pixmap)
        {
                w->pixmap = XCompositeNameWindowPixmap (dpy, w->id);
-               w->glxPixmap = glXCreatePixmap (dpy, w->fbConfig, w->pixmap, w->isOverlay ? tfpAttribsRGBA : tfpAttribs);
+               w->glxPixmap = glXCreatePixmap (dpy, w->fbConfig, w->pixmap, tfpAttribs);
                
                glBindTexture (GL_TEXTURE_2D, w->texName);
                __pointer_to_glXBindTexImageEXT (dpy, w->glxPixmap, GLX_FRONT_LEFT_EXT, NULL);
```

Essentially, I had to turn off the alpha channel of the overlay. This leads to various issues like no transparency, notifications with black borders, and the on-screen keyboard not being usable.

I also noticed that there is an X11 error happening: "BadDrawable (invalid Pixmap or Window parameter)".
I haven't been able to narrow down what is causing it or even whether it is related to this issue though.

**Comment by Plagman** [comment]
Overlay windows are depth 32 now, so this is likely related to that change. It should be more correct than before, but maybe there's a slight attribute mismatch in the glX operation to account for now.

**Comment by alkazar** [comment]
Thanks for that info. I suspected there was some kind of pixel format change to the overlay, but what was throwing me off was that everything is working fine on AMD GPUs.

**Comment by alkazar** [comment]
@Plagman I believe I know what the problem is. I think it is either an issue in the Steam client in the way that it selects X visuals for the overlay or possibly the Nvidia drivers in the way that it reports X visuals.

After some debugging, this is what I have found:
 - In `win_fbconfig` the X visual attached to the overlay window has a reported depth of 32, BUT the actual depth buffer, as reported by `glxinfo` for the X visual id, is 24-bit and it does not have an alpha channel.
 - The selected X visual appears to be the first in the list that has a depth of 32, perhaps the Steam client is just picking the first X visual with a reported depth of 32, and assumes that is correct without checking further.
 - This results in `win_fbconfig` selecting a `GLXFBConfig` object without an alpha channel. This also explains the X11 `BadDrawable` error and explains why the overlay works (sans transparency) when I override the parameters for `glXCreatePixmap` to use RGB instead of RGBA, because the window is actually 24-bit and doesn't have alpha.
 - On my AMD machine, all X visuals with a reported depth of 32 also have an actual depth buffer that is 32-bit. Explaining why it works in the AMD environment.

TLDR: the way the Nvidia driver reports bit depth is a bit wonky and I believe the Steam client is not accounting for it and inadvertently selecting the wrong resource for a 32-bit window.

I don't think there is anything that can be done in the SteamOS compositor to fix this. It most likely needs to be fixed in the Steam client. X11 doesn't like it when you select a `GLXFBConfig` with a different X visual id from the current window.

Not sure how to proceed in getting this looked at and addressed. Please advise!
This is quite painful for all SteamOS compositor users.

Thanks!

**Comment by oolongbrothers** [comment]
Hello everybody!

I am seeing the same behaviour on SteamOS 2.0 with Nvidia GPU since the June 1st Steam client update.

Interestingly, this does **only appear when I use the SteamOS compositor** session. When I change to a Gnome session and run the Steam client in Big Picture more, the overlay works as expected.

Any progress on this would be highly appreciated since it blocks much of my and potentially others' SteamOS usage. I can play using the Gnome session but Gnome's heavy screen tearing issues make it quite unbearable for most content.

Kind regards.

**Comment by Mohandevir** [comment]
Same thing for me. I'm using the Steamos-Compositor on top of Mint 20.

GTX 1660 Super with driver 440.100

Thanks for your support!

**Comment by Wazz4657** [comment]
Same issues here:

GTX 1660ti / SteamOS & GamerOS

**Comment by alkazar** [comment]
I found a work around.

For SteamOS/steamos-compositor:
Edit `/usr/bin/steamos-session` by removeing the `-steamos` launch option to steam.

For GamerOS/steamos-compositor-plus:
Edit `~/.config/steamos-compositor-plus`, adding the following line:
`STEAMCMD="steam -tenfoot -fulldesktopres"`

The downside of this change is that you will:
 - lose access to audio, bluetooth and network settings
 - have a bunch of irrelevant extra options in the shutdown menu
 - lose access to OS version information
 - likely lose access to OS update notifications

**Comment by abc-mikey** [comment]
Since this is affecting SteamOS users I'm surprised valve haven't taken steps to fix this already. I'm not sure if there are any steps to be taken to wake them up.

**Comment by andy3471** [comment]
@alkazar Thanks for the workaround. The loss of config is annoying, but it's better than no overlay. Looks like I still got a gameros update notification with this set like that. Hopefully someone at valve picks this up as this basically makes steamos unusable for nvidia users

**Comment by Wazz4657** [comment]
Thanks @alkazar this was a massive annoyance

**Comment by Plagman** [comment]
Without -steamos you also won't get the 1080p/60 overlay regardless of the game's resolution, and you won't get the overlay if the game hangs or locks up to be able to exit it, which can be pretty important. I'm looking into the issue now and hopefully we can resolve it soon.

**Comment by alkazar** [comment]
@Plagman Thanks for looking into this! In case it helps, in addition to my post above, which I think is still accurate as far as my understanding goes, I tried the following:
 - I tried forcing use of a different X visual which has an alpha channel, but that just resulted in artifacting and general badness
 - I tried cycling through dozens of different `GLXFBConfig`s, with no result

I also noticed the following:
 - the X visual being used in Gnome was the same one used under the steamos compositor i.e. one without proper alpha support according to `glxinfo`
 - the overlay seems to correctly have transparency when run with the `-steamos` option under Gnome, which seems to conflict with the above point
 - in `win_fbconfig` there is only a single `GLXFBConfig` that matches the criteria of having the same `visualid` and it doesn't have an alpha channel

**Comment by kisak-valve** [comment]
Hello, per "Fixed a regression with transparent visual selection for SteamOS overlays and on-screen keyboard causing invalid rendering on NVIDIA configurations" in the [2020-08-05 steam client beta update](https://steamcommunity.com/groups/SteamClientBeta/announcements/detail/2735326880599543430), please opt into Steam's beta client and retest.

**Comment by alkazar** [comment]
Fix confirmed on Arch Linux when using `steamos-compositor-plus` and a GTX 670. Also confirmed no regression to the overlay when running under Gnome.

Thank you for your efforts @Plagman and @kisak-valve !

**Comment by brain-anti-freeze** [comment]
Confirmed working now on Arch with steamos-compositor-plus and GTX2070 (450.56.02).  Thanks for the support !

**Comment by afonit** [comment]
Confirmed working on SteamOS beta using Nvidia GeForce GTX 960

**Comment by Wazz4657** [comment]
Working on GamerOS 19 / GTX 1660 TI


THANKS!!

**Comment by kisak-valve** [comment]
Closing as fixed per "Fixed a regression with transparent visual selection for SteamOS overlays and on-screen keyboard causing invalid rendering on NVIDIA configurations" in the [2020-08-29](https://store.steampowered.com/news/74990/) Steam client update.

**Comment by Alia5** [comment]
This happens to me with the latest Steam beta

...with an **AMD GPU**   (5700xt)  
...on **Windows**  
🤦‍♂️
